spec/migrations/update-6_0_0/index.ts

```diff
@@ -0,0 +1,87 @@
+import { Rule, SchematicContext, Tree, SchematicsException, chain,  } from '@angular-devkit/schematics';
+import { TsLintTask, NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+
+export function rxjsV6MigrationSchematic(_options: any): Rule {
+  return chain([
+    addDependencies(),
+    runTsLint(),
+  ]);
+}
+
+const rxjsCompatVersion = '^6.0.0-rc.0';
+const rxjsTSLintVersion = '0.0.0';
+
+function addDependencies() {
+  return (tree: Tree, context: SchematicContext) => {
+      const pkgPath = '/package.json';
+      const buffer = tree.read(pkgPath);
+      if (buffer == null) {
+        throw new SchematicsException('Could not read package.json');
+      }
+      const content = buffer.toString();
+      const pkg = JSON.parse(content);
+
+      if (pkg === null || typeof pkg !== 'object' || Array.isArray(pkg)) {
+        throw new SchematicsException('Error reading package.json');
+      }
+
+      if (!pkg.dependencies) {
+        pkg.dependencies = {};
+      }
+      if (!pkg.devDependencies) {
+        pkg.devDependencies = {};
+      }
+
+      pkg.dependencies['rxjs-compat'] = rxjsCompatVersion;
+      pkg.devDependencies['rxjs-tslint'] = rxjsTSLintVersion;
+
+      tree.overwrite(pkgPath, JSON.stringify(pkg, null, 2));
+      context.addTask(new NodePackageInstallTask());
+
+      return tree;
+  };
+}
+
+function runTsLint() {
+  return (tree: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(tree);
```

Comment:
This limits the update schematic to angular CLI projects. I'm pretty sure you don't need this restriction; you should be able to run tslint over the root tsconfig (or all TS files). We have a Tslint task now.